cpp/dolfinx/fem/assemble_matrix_impl.h

```diff
@@ -343,8 +343,19 @@ void assemble_interior_facets(
     kernel(Ae.data(), coeff_array.data(), constants.data(),
            coordinate_dofs.data(), local_facet.data(), perm.data());
 
-    dof_transform(Ae, cell_info, cells[0], num_cols);
-    dof_transform_to_transpose(Ae, cell_info, cells[0], num_rows);
+    const xtl::span<T> _Ae(Ae);
+
+    const xtl::span<T> sub_Ae
+        = _Ae.subspan(bs0 * dmap0_cell0.size() * num_cols,
+                      bs0 * dmap0_cell1.size() * num_cols);
+    const xtl::span<T> sub_Ae2
+        = _Ae.subspan(bs1 * dmap1_cell0.size(),
+                      num_rows * num_cols - bs1 * dmap1_cell0.size());
+
+    dof_transform(_Ae, cell_info, cells[0], num_cols);
```

Comment:
Could you add a comment? It's not immediately clear to me why this sequence of transforms are required.